pykusto/_src/query.py

```diff
@@ -476,18 +477,18 @@ def _compile(self) -> KQL:
 class _OrderQueryBase(Query):
     class OrderSpec:
         col: OrderedType
-        order: Order
-        nulls: Nulls
+        order: Optional[Order]
+        nulls: Optional[Nulls]
 
-        def __init__(self, col: OrderedType, order: Order, nulls: Nulls):
+        def __init__(self, col: OrderedType, order: Optional[Order], nulls: Optional[Nulls]):
             self.col = col
             self.order = order
             self.nulls = nulls
 
     _query_name: str
     _order_specs: List[OrderSpec]
 
-    def __init__(self, head: Query, query_name: str, col: OrderedType, order: Order, nulls: Nulls):
+    def __init__(self, head: Query, query_name: str, col: OrderedType, order: Optional[Order], nulls: Optional[Nulls]):
```

Comment:
See previous responses about the meaning of `Optional` and implicit `Optional`.

Reply:
If we want to enforce the user to explicit supply `None`, then the `then_by` function doesn't need default arguments no?

Reply:
I don't want to force the user to explicitly provide `None`, that would be inconvenient. But `_OrderQueryBase.__init__` is not called by the user, it's called internally, and this is where I want to force an explicit `None`. Or to look at it differently, there is nothing to gain by making the arguments optional here, since all the arguments need to be provided anyway (look at the call to this method in `_SortQuery.__init__`).